### PR TITLE
http port bug fix

### PR DESCRIPTION
**Pull request by initard**
## Proposed changes

Adding http port to `tedge config list` command

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by initard** [comment]
Hold up, I need to investigate the commits. There should only be one commit.

**Comment by initard** [comment]
Ready for review

File: crates/core/tedge/src/cli/config/config_key.rs [hunk 1]
```diff
@@ -51,6 +51,7 @@ impl ConfigKey {
             config_key!(AzureMapperTimestamp),
             config_key!(MqttBindAddressSetting),
             config_key!(MqttPortSetting),
+            config_key!(HttpPortSetting),
```
Comment (didier-wenzek) [inline crates/core/tedge/src/cli/config/config_key.rs:54]: You can also add `HttpBindAddressSetting`, making this setting read only as for `DeviceIdSetting`.

Comment (initard) [reply crates/core/tedge/src/cli/config/config_key.rs:55]: Done

**Review by didier-wenzek — CHANGES_REQUESTED** [review]
Adding `HttpBindAddressSetting` to `tedge config list` is not enough.

What is proposed here would be misleading with  `tedge config get` returning a value that is not what has been `tedge config set`.

One has:
* either to make `HttpBindAddressSetting` readonly.
* or to use the value set with `tedge config set`.

I'm for the former (as this is what is requested by https://github.com/thin-edge/thin-edge.io/issues/1479.

```
impl ConfigSettingAccessor<HttpBindAddressSetting> for TEdgeConfig {
    fn query(&self, _setting: HttpBindAddressSetting) -> ConfigSettingResult<IpAddress> {
        // Use MQTT address !!!
        ...
    }

    fn update(
        &mut self,
        _setting: HttpBindAddressSetting,
        value: IpAddress,
    ) -> ConfigSettingResult<()> {
        Err(ConfigSettingError::ReadonlySetting { ... })
    }

    fn unset(&mut self, _setting: HttpBindAddressSetting) -> ConfigSettingResult<()> {
        Err(ConfigSettingError::ReadonlySetting { ... })
    }
}
```

For details, see the implementation of `ConfigSettingAccessor<DeviceIdSetting>`

File: crates/core/tedge/src/cli/config/config_key.rs (hunk 1 above)
Comment (albinsuresh) [reply crates/core/tedge/src/cli/config/config_key.rs:55]: Why read-only? We don't want to let users configure the `HttpBindAddress`? If not, why are we even adding it?

Comment (didier-wenzek) [reply crates/core/tedge/src/cli/config/config_key.rs:55]: @albinsuresh and @reubenmiller 

> Why read-only? We don't want to let users configure the `HttpBindAddress`? If not, why are we even adding it?

To be aligned with https://github.com/thin-edge/thin-edge.io/issues/1479 where we conclude that for now we will use the MQTT bind address for both HTTP and MQTT.

We just need to be consistent here. I see 3 options:

* either hide `HttpBindAddress` to the user (i.e not listed by `tedge config list`).
* or show `HttpBindAddress` but read only,  to let the user know that this is using the MQTT setting .
* or make `HttpBindAddress` an independent setting the user can list,get and set.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**